cli/azd/internal/appdetect/spring_boot_property_test.go

```diff
@@ -0,0 +1,70 @@
+package appdetect
+
+import (
+	"github.com/stretchr/testify/require"
+	"os"
+	"path/filepath"
+	"testing"
+)
+
+func TestReadProperties(t *testing.T) {
+	var properties = readProperties(filepath.Join("testdata", "java-spring", "project-one"))
+	require.Equal(t, "", properties["not.exist"])
+	require.Equal(t, "jdbc:h2:mem:testdb", properties["spring.datasource.url"])
+
+	properties = readProperties(filepath.Join("testdata", "java-spring", "project-two"))
+	require.Equal(t, "", properties["not.exist"])
+	require.Equal(t, "jdbc:h2:mem:testdb", properties["spring.datasource.url"])
+
+	properties = readProperties(filepath.Join("testdata", "java-spring", "project-three"))
+	require.Equal(t, "", properties["not.exist"])
+	require.Equal(t, "HTML", properties["spring.thymeleaf.mode"])
+
+	properties = readProperties(filepath.Join("testdata", "java-spring", "project-four"))
+	require.Equal(t, "", properties["not.exist"])
+	require.Equal(t, "mysql", properties["database"])
```

Comment:
This can be improved, just like the `TestGetEnvironmentVariablePlaceholderHandledValue` in the same file. But to make the PR easier to review, let's change it in another PR.